### PR TITLE
Fix input typing by dispatching key events instead of adb input text

### PR DESCRIPTION
**Pull request by amanjeetsingh150**
## Proposed Changes
1. Automated copying of maestro app and maestro server apks whenever following tasks are executed:

```
./gradlew :maestro-android:assembleAndroidTest
./gradlew :maestro-android:assemble
```
PS: Also includes keystore in repo root 

2. Use UI Automator for dispatching key events for typing instead of adb input text.

## Testing
Tested this combination for being exhaustive: 
```
QWERTY qwerty 12345 ~!@#$%^&*() +-/* `\';,./ {}|:"<>?
```
[device-2022-11-17-181258.webm](https://user-images.githubusercontent.com/12881364/202449921-bf972c88-d74f-43cd-a643-3325a68304d5.webm)

## Issues Fixed
https://github.com/mobile-dev-inc/maestro/issues/395

### CODE REVIEW

File: maestro-android/src/androidTest/java/dev/mobile/maestro/MaestroDriverService.kt [hunk 1]
```diff
@@ -103,4 +128,71 @@ class Service(
...
+                    uiDevice.pressKeyCode(element.code - 36, 1)
+                }
+                in 97..121 -> {
+                    /** a~z **/
+                    uiDevice.pressKeyCode(element.code - 68)
+                }
+                ';'.code -> uiDevice.pressKeyCode(KEYCODE_SEMICOLON)
+                '='.code -> uiDevice.pressKeyCode(KEYCODE_EQUALS)
+                ','.code -> uiDevice.pressKeyCode(KEYCODE_COMMA)
+                '-'.code -> uiDevice.pressKeyCode(KEYCODE_MINUS)
```
Comment (dmitry-zaitsev) [inline maestro-android/src/androidTest/java/dev/mobile/maestro/MaestroDriverService.kt:162]: Let's add underscore as well

File: maestro-android/build.gradle [hunk 2]
```diff
@@ -58,6 +66,43 @@ android {
...
+    it.doLast {
+        if (!layout.buildDirectory.file(maestroServerApkDestPath).get().asFile.exists())
+            throw new GradleException("Error: Input source for copyMaestroServer doesn't exist")
+
+        new File("./maestro-client/src/main/resources/maestro-android-debug-androidTest.apk")
+            .renameTo(new File("./maestro-client/src/main/resources/maestro-server.apk"))
+    }
+}
+
+tasks.named("assemble") { finalizedBy("copyMaestroAndroid") }
```
Comment (dmitry-zaitsev) [inline maestro-android/build.gradle:103]: Nice!

**Review by dmitry-zaitsev — APPROVED** [review]

**Comment by Leland-Takamine** [comment]
@amanjeetsingh150 Is this ready to land?

**Comment by amanjeetsingh150** [comment]
> @amanjeetsingh150 Is this ready to land?

Merging this now

**Pull request merged**